jme3-core/src/main/java/com/jme3/math/Rectangle.java

```diff
@@ -125,6 +137,35 @@ public void setC(Vector3f c) {
         this.c = c;
     }
 
+    /**
+     * <code>getD</code> returns the fourth point of the rectangle.
+     * The fourth point is given by calculating (B + C) - A.
+     *
+     * @return the fourth point of this rectangle.
+     */
+    public Vector3f getD() {
+        return b.add(c).subtractLocal(a);
+    }
+
+    /**
+     * Computes the normal vector of this <code>Rectangle</code>.
+     * 
+     * @param normal the vector to store the result in
+     * @return the normal vector of this <code>Rectangle</code>
+     * @see https://web.ma.utexas.edu/users/m408m/Display12-5-4.shtml

```

Comment:
The correct syntax for inserting an external link in the javadoc should be this: @see <a href="URL#value"> label </a>. Despite this, we are sure that it is useful to insert a link to a domain outside the perimeter of the jMonkey engine that could be removed at any time or not accessible for all countries? If the page were suddenly removed we would have a dead link in the javadoc.

Reply:
I would suggest creating a specific test case for this class such as `TestRecatangleMesh` and not modifying the current test cases, where perhaps it would be better to replace the `Quad` with the `CenterQuad`, keeping the creation of floors simple and intuitive for new users. Just my opinion.

Reply:
Link removed.

Reply:
> perhaps it would be better to replace the `Quad` with the `CenterQuad`, keeping the creation of floors simple and intuitive for new users. Just my opinion.

For floors, `RectangleMesh` is more intuitive because it doesn't require rotation of the `Geometry`. `CenterQuad` is in the X-Y plane, so it requires rotation to create an X-Z floor.